### ISSUE TITLE
zperf: Concurrency improvements

### ISSUE THREAD

**Issue by krish2718** [zephyrproject-rtos/zephyr#55963]
Zperf tool should help to closely emulate application usecases for testing and troubleshooting. With recent addition of setting TOS, we can run two Zperf sessions in parallel (1st - async, 2nd - sync), the async one is run using a workqueue whose priority is not user configurable and sync one uses the shell thread.

We could implement below:

1. Run each session using a dedicated thread, makes way for concurrent sessions
2. Each thread priority should be configurable using both Kconfig and also the Zperf API/CLI. This is necessary to make the thread priority according to its ToS value.
3. Extend to run concurrent async sessions in parallel

Usecase:

Run traffic using 3 zperf sessions with TOS = 0x00, 0xA0, 0xE0 in parallel with corresponding thread priorities (else the throughput differentiation won't happen).

**Comment by zephyrbot** [comment]
Hi @nashif,

This issue, marked as an Enhancement, was opened a while ago and did not get any traction. It was just assigned to you based on the labels. If you don't consider yourself the right person to address this issue, please re-assing it to the right person.

Please take a moment to review if the issue is still relevant to the project.  If it is, please provide feedback and direction on how to move forward. If it is not, has already been addressed, is a duplicate, or is no longer relevant, please close it with a short comment explaining the reason.


@krish2718 you are also encouraged to help moving this issue forward by providing additional information and confirming this request/issue is still relevant to you.

Thanks!
